src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -249,81 +315,507 @@ public DateTime(int year, int month, int day, int hour, int minute, int second,
+        /// <param name="second">The seconds (0 through 59).</param>
+        /// <param name="millisecond">The milliseconds (0 through 999).</param>
+        /// <param name="microsecond">The milliseconds (0 through 900).</param>
+        /// <param name="calendar">The calendar that is used to interpret <paramref name="year"/>, <paramref name="month"/>, and <paramref name="day"/>.</param>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="calendar"/> is <see langword="null"/>
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// <paramref name="year"/> is not in the range supported by <paramref name="calendar"/>.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="month"/> is less than 1 or greater than the number of months in <paramref name="calendar"/>.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="day"/> is less than 1 or greater than the number of days in <paramref name="month"/>.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="hour"/> is less than 0 or greater than 23.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="minute"/> is less than 0 or greater than 59.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="second"/> is less than 0 or greater than 59.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="millisecond"/> is less than 0 or greater than 999.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="microsecond"/> is less than 0 or greater than 900.
+        /// </exception>
+        /// <remarks>
+        /// The allowable values for <paramref name="year"/>, <paramref name="month"/>, and <paramref name="day"/> parameters
+        /// depend on the <paramref name="calendar"/> parameter. An exception is thrown if the specified date and time cannot
+        /// be expressed using <paramref name="calendar"/>.
+        ///
+        /// For applications in which portability of date and time data or a limited degree of time zone awareness is important,
+        /// you can use the corresponding <see cref="DateTimeOffset"/> constructor.
+        /// </remarks>
+        public DateTime(int year, int month, int day, int hour, int minute, int second, int millisecond, int microsecond, Calendar calendar!!)
+        {
             if (second != 60 || !s_systemSupportsLeapSeconds)
             {
-                ulong ticks = DateToTicks(year, month, day) + TimeToTicks(hour, minute, second);
-                ticks += (uint)millisecond * (uint)TicksPerMillisecond;
-                Debug.Assert(ticks <= MaxTicks, "Input parameters validated already");
-                _dateData = ticks;
+                var dateTime = calendar.ToDateTime(year, month, day, hour, minute, second, millisecond);
+                dateTime = dateTime.AddMicroseconds(microsecond);
```

Comment:
>dateTime = dateTime.AddMicroseconds(microsecond); [](http://example.com/codeflow?start=16&length=49)

This is wrong. imaging users passing microseconds equal -10 for example. this should throw but with this code possibly will not throw. 

You need to manually test microseconds, like:

```C#

if ((uint)microsecond >= MicrosecondsPerMillisecond) 
{
    Throw...
}

var dateTime = calendar.ToDateTime(year, month, day, hour, minute, second, millisecond);
_dateData = dateTime.AddMicroseconds(microsecond)._dateData; // Note, it is wrong to assign _dateData to UTicks.
```